### PR TITLE
support authz for alertmanager.

### PR DESCRIPTION
**Pull request by morvencao**
ref: https://github.com/open-cluster-management/backlog/issues/12110
depends on: open-cluster-management/observatorium-operator#26

Hub's side change for Alertmanager.

Signed-off-by: morvencao <lcao@redhat.com>

### CODE REVIEW

**Comment by morvencao** [comment]
/hold 
Depends on https://github.com/open-cluster-management/observatorium-operator/pull/26

File: manifests/base/thanos/thanos-ruler-clusterrole.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+kind: ClusterRole
+metadata:
+  name: open-cluster-management:thanos-rule
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - "*"
```
Comment (clyang82) [inline manifests/base/thanos/thanos-ruler-clusterrole.yaml:11]: Just `Get` permission?

Comment (morvencao) [reply manifests/base/thanos/thanos-ruler-clusterrole.yaml]: Done.

File: pkg/config/config.go [hunk 2]
```diff
@@ -95,6 +104,11 @@ const (
 	ConfigmapReloaderImgTagSuffix = "4.5.0"
 	ConfigmapReloaderKey          = "prometheus-config-reloader"
 
+	OauthProxyImgRepo      = "quay.io/openshift"
+	OauthProxyImgName      = "origin-oauth-proxy"
+	OauthProxyImgTagSuffix = "4.5.0"
+	OauthProxyKey          = "alertmanager-oauth-proxy"
```
Comment (clyang82) [inline pkg/config/config.go:110]: it should be `oauth_proxy` which exists in `oc get cm -n open-cluster-management -oyaml mch-image-manifest-2.3.0`

Comment (morvencao) [reply pkg/config/config.go]: good catch! Addressed.
I found that `oauth-proxy` image only exists in `mch-image-manifest` configmap and the not always built and tagged. I need to handle it specially.

File: controllers/placementrule/hub_info_secret.go [hunk 3]
```diff
@@ -38,9 +39,15 @@ func newHubInfoSecret(client client.Client, obsNamespace string,
...
 	}
+	hubRouterCA, err := config.GetRouterCA(client)
+	if err != nil {
+		log.Error(err, "Failed to CA of openshift Route")
+		return nil, err
+	}
 	hubInfo := &HubInfo{
 		ClusterName: clusterName,
 		Endpoint:    url + urlSubPath,
+		HubRouterCA: hubRouterCA,
```
Comment (marcolan018) [inline controllers/placementrule/hub_info_secret.go:50]: do we also need to add the url for alertmanagaer route here?

Comment (morvencao) [reply controllers/placementrule/hub_info_secret.go:50]: yes, we can.

Comment (morvencao) [reply controllers/placementrule/hub_info_secret.go]: added.

**Comment by morvencao** [comment]
Depends on https://github.com/open-cluster-management/observability-e2e-test/pull/177

**Pull request merged**